pkg/bpf/map_linux.go

```diff
@@ -836,22 +888,28 @@ func (m *Map) Update(key MapKey, value MapValue) error {
 	m.lock.Lock()
 	defer m.lock.Unlock()
 
+	defer m.updatePressureMetric()
```

Comment:
I'd guess we only need to update this metric if `err != nil`, any particular reason not to have it inside the same defer function below which already has checks for whether the cache is enabled and checks against the `err`?

Reply:
I removed the unnecessary `defer` and added `updatePressureMetric()` to the existing func. Still called when `err != nil` and caching enabled since even in that case an entry is added to the cache.
Also, I was calling `updatePressureMetric()` pretty liberally due to its idempotency but I tried to tweak better the places where it is called.